src/app/dataset/dataset.component.ts

```diff
@@ -366,16 +366,27 @@ export class DatasetComponent implements OnInit, OnDestroy {
     );
   }
 
-  /** publicationFitnessWarning
-  /* - return relevant warning message according to the status
+  /** publicationFitnessWarningAndClass
+  /* - return object literal specifying the relevant warning message and css class for the given status
   /* @param {string} status - the publication status
   */
-  publicationFitnessWarning(status?: string): string {
+  publicationFitnessWarningAndClass(
+    status?: string
+  ): { warning: string; cssClass: string } | undefined {
+    if (status === PublicationFitness.FIT) {
+      return undefined;
+    }
+    let cssClass = '';
+    let warning = '';
     if (status === PublicationFitness.UNFIT) {
-      return this.translate.instant('datasetUnpublishableBanner');
+      warning = this.translate.instant('datasetUnpublishableBanner');
     } else if (status === PublicationFitness.PARTIALLY_FIT) {
-      return this.translate.instant('datasetPartiallyUnpublishableBanner');
+      warning = this.translate.instant('datasetPartiallyUnpublishableBanner');
+      cssClass = 'partial-fitness';
```

Comment:
Perhaps more consistent with the rest of the code to use the translation option as a pipe in the html document?